src/Core/Domain/Discount/QueryResult/DiscountForEditing.php

```diff
@@ -112,4 +114,14 @@ public function getType(): DiscountType
     {
         return new DiscountType($this->type);
     }
+
+    public function getPercentDiscount(): ?float
```

Comment:
```suggestion
    public function getPercentDiscount(): ?DecimalNumber
```
We favor `DecimalNumber` in all CQRS objects in place of `float` Always the same reason, float are unreliable
This is one of the few exceptions where we accept an object instead of scalar values in CQRS objects (especially in queries)